amaranth/vendor/gowin.py

```diff
@@ -0,0 +1,549 @@
+from abc import abstractproperty
+
+from ..hdl import *
+from ..lib.cdc import ResetSynchronizer
+from ..build import *
+
+# Acknowledgments:
+#   Parts of this file originate from https://github.com/tcjie/Gowin
+
+
+__all__ = ["GowinPlatform"]
+
+
+class GowinPlatform(TemplatedPlatform):
+    """
+    .. rubric:: Apicula toolchain
+
+    Required tools:
+        * ``yosys``
+        * ``nextpnr-gowin``
+        * ``gowin_pack``
+
+    The environment is populated by running the script specified in the environment variable
+    ``AMARANTH_ENV_Apicula``, if present.
```

Comment:
In `main` branch this should be `AMARANTH_ENV_APICULA`.

Reply:
Should the toolchain name be uppercase? Or just uppercase in the _toolchain_env_var?

Reply:
Only in the environment variable name. I.e. only the docs need to be changed.

Reply:
Now it is all caps.